### PR TITLE
Feature: Accept PCMK_OCF_DEGRADED and PCMK_OCF_DEGRADED_MASTER codes

### PR DESCRIPTION
**Pull request by oalbrigt**
also added OCF_CHECK_LEVEL to Dummy agent to return custom rc.

### CODE REVIEW

File: extra/resources/Dummy [hunk 1]
```diff
@@ -221,6 +221,12 @@ dummy_monitor() {
             return $OCF_ERR_GENERIC
             ;;
 
+            40)
+            # monitor level that returns error code from state file
+            rc=$(cat ${OCF_RESKEY_state})
+            [ -n "$rc" ] && ocf_exit_reason "return code from ${OCF_RESKEY_state}" && return $rc
```
Comment (kgaillot) [inline extra/resources/Dummy:227]: You can come up with a better message than this ;) It's a dummy resource for testing, so we can have a little fun with it

File: include/crm/common/ocf.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    PCMK_OCF_EXEC_ERROR    = 192, /* Generic problem invoking the agent */
+    PCMK_OCF_UNKNOWN       = 193, /* State of the service is unknown - used for recording in-flight operations */
+    PCMK_OCF_SIGNAL        = 194,
+    PCMK_OCF_NOT_SUPPORTED = 195,
+    PCMK_OCF_PENDING       = 196,
+    PCMK_OCF_CANCELLED     = 197,
+    PCMK_OCF_TIMEOUT       = 198,
+    PCMK_OCF_OTHER_ERROR   = 199, /* Keep the same codes as PCMK_LSB */
+};
```
Comment (kgaillot) [inline include/crm/common/ocf.h:31]: Let's put the enum move in its own commit. I hate to realize this after you've done all the work, but it now occurs to me that include/crm/common/results.h would be the perfect place for this enum.

File: lib/common/agents.c [hunk 3]
```diff
@@ -67,6 +67,27 @@ pcmk_get_ra_caps(const char *standard)
...
+            remapped_rc = PCMK_OCF_RUNNING_MASTER;
+            break;
+
+        default:
+            break;
+    }
+
+    return remapped_rc;
+}
+
```
Comment (kgaillot) [inline lib/common/agents.c:90]: We can also replace the existing switch in remap_monitor_rc() with a call to the new function, to reduce code duplication

**Review by kgaillot — COMMENTED** [review]
Looking good.

For a scheduler regression test, you'll want to create a test cluster with a primitive resource that returns degraded, and a promotable clone that returns degraded master (on the master). Then find the "saving inputs" log message on the DC that corresponds to when those values were recorded. You can verify by looking in the file for lrm_rsc_op entries with operation="monitor" and rc-code= 190 or 191. That file can be used as the test input.

File: include/crm/common/ocf.h (hunk 2 above)
Comment (oalbrigt) [reply include/crm/common/ocf.h]: No worries :)

**Comment by kgaillot** [comment]
Awesome, thanks
